fermenter-mda/src/main/java/org/technologybrewery/fermenter/mda/notification/NotificationService.java

```diff
@@ -69,56 +82,58 @@ static void addNotificationToPassedMap(String targetFile, Notification notificat
     }
 
     /**
-     * Merges any notifications from the current plugin execution into the session for storage across plugin invocations.
+     * Writes any encountered notifications to a file for use later.  These are written to
+     * /target/manual-action-notifications.
      */
-    public synchronized void mergeNotificationsIntoSessionForCrossProjectStorage() {
+    public void recordNotifications() {
+        int manualActionCount = 0;
+        File projectParentFile = new File(NOTIFICATION_DIRECTORY_PATH);
         Map<String, Map<String, Notification>> collectorNotifications = NotificationCollector.getNotifications();
-        Map<String, Map<String, Notification>> sessionNotifications = getNotificationsFromSession();
-
-        synchronized (this) {
-            if (sessionNotifications == null) {
-                sessionNotifications = collectorNotifications;
-                Properties userProperties = session.getUserProperties();
-                userProperties.put(NOTIFICATIONS, sessionNotifications);
-            } else {
-                for (Map.Entry<String, Map<String, Notification>> entry : collectorNotifications.entrySet()) {
-                    String fileName = entry.getKey();
-                    Map<String, Notification> notificationsForFile = entry.getValue();
-                    for (Map.Entry<String, Notification> subMapEntry : notificationsForFile.entrySet()) {
-                        addNotificationToPassedMap(fileName, subMapEntry.getValue(), sessionNotifications);
-                    }
+        for (Map.Entry<String, Map<String, Notification>> entry : collectorNotifications.entrySet()) {
+            String fileName = entry.getKey();
+            Map<String, Notification> notificationsForFile = entry.getValue();
+            int i = 0;
+            for (Map.Entry<String, Notification> subMapEntry : notificationsForFile.entrySet()) {
+                File outputFile = new File(projectParentFile, fileName + "-" + i++ + ".txt");
+                try {
+                    FileUtils.forceMkdir(projectParentFile);
+                    FileUtils.write(outputFile, subMapEntry.getValue().getNotificationAsString(), Charset.defaultCharset());
```

Comment:
A: Minor, but java.nio.Files can do both of these things without a 3rd party lib.

Reply:
Fair, though this library is already there, so feels like 6 and one half